src/wasm/wasm-type.cpp

```diff
@@ -746,6 +755,26 @@ HeapType& HeapType::operator=(const HeapType& other) {
   return *this;
 }
 
+bool Field::operator<(const Field& other) const {
+  if (type < other.type) {
+    return true;
+  }
+  if (other.type < type) {
+    return false;
+  }
+  if (mutable_ < other.mutable_) {
+    return true;
+  }
+  if (other.mutable_ < mutable_) {
+    return false;
+  }
```

Comment:
This one's slightly weirder but should still work:
```suggestion
  if (mutable_ != other.mutable_) {
    return mutable < other.mutable;
  }
```
It also might be worth comparing the type last to avoid having to compare types in as many cases as possible.